src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -212,39 +211,26 @@ protected override DbParameter CreateDbParameter()
         /// </summary>
         public override void Prepare()
         {
-            if (Connection == null
-                || Connection.State != ConnectionState.Open)
+            if (_connection?.State != ConnectionState.Open)
             {
                 throw new InvalidOperationException(Resources.CallRequiresOpenConnection(nameof(Prepare)));
             }
 
-            if (string.IsNullOrEmpty(CommandText))
+            if (string.IsNullOrEmpty(_commandText))
             {
                 throw new InvalidOperationException(Resources.CallRequiresSetCommandText(nameof(Prepare)));
             }
 
-            if (_preparedStatements.Count == 0)
+            if (_preparedStatements.Count != 0)
             {
-                var tail = CommandText;
-                sqlite3_stmt stmt;
-                do
-                {
-                    (stmt, tail) = GetPreparedStatement(tail);
-
-                    // Statement was empty, white space, or a comment
-                    if (stmt.ptr == IntPtr.Zero)
-                    {
-                        if (!string.IsNullOrEmpty(tail))
-                        {
-                            continue;
-                        }
-
-                        break;
-                    }
+                return;
+            }
 
-                    _preparedStatements.Add(stmt);
+            using (var enumerator = PrepareAndEnumerateStatements().GetEnumerator())
```

Comment:
It would work. This just avoid having a local variable for the current value. (It's the same thing we do for `.Load()`.)